test/config/utility.cc

```diff
@@ -73,7 +73,22 @@ std::string ConfigHelper::baseConfigNoListeners() {
                      Platform::null_device_path, Platform::null_device_path);
 }
 
-std::string ConfigHelper::baseConfig() {
+std::string ConfigHelper::baseConfig(bool multiple_addresses) {
+  if (multiple_addresses) {
+    return absl::StrCat(baseConfigNoListeners(), R"EOF(
+  listeners:
+  - name: listener_0
+    address:
+      socket_address:
+        address: 127.0.0.1
+        port_value: 0
+    additional_addresses:
```

Comment:
I added a flag `multiple_addresses` to the `baseConfig()`, the default value is false:
https://github.com/envoyproxy/envoy/pull/22081/files#diff-a00c6efc42fc025c0cea76be1b3c81c6993b65edb65c77e5001ed22157236386R157

only set it as true for the multiple addresses tests, like:
https://github.com/envoyproxy/envoy/pull/22081/files#diff-7fe1ad3a088b5d0ea44d9bace0dfa0675d5009bcb538afb778ebad50c802ddeaR231

so it won't affect any existing test.

I also checked, we will rename the listener to `http` for legacy test
https://github.com/envoyproxy/envoy/blob/cb78092e9219da3fd77e49aa1273435696c72921/test/integration/http_integration.cc#L327-L329

With this multiple addresses test, the `http` port still points to the first address. But for the multiple addresses test, it will register ports name by itself, like
https://github.com/envoyproxy/envoy/pull/22081/files#diff-7fe1ad3a088b5d0ea44d9bace0dfa0675d5009bcb538afb778ebad50c802ddeaR434

so all the multiple addresses tests won't depend on `http` port.

But yes, I didn't have a strong reason to change `baseConfig()`, I just feel it is easier and readable to write a yaml config than the modifier.

Let me know if you think changing the base config is still good choice.

Reply:
ehh, I prefer to not have 8 ways to set configuration but I'll accept your arguments for readability given my main concern (that it not affect other tests) is as you point out mistaken, so let's leave as-is :-)
I'll defer to Dan for QUIC comments, then do final pass.